test/e2e/tests/infra/infra.go

```diff
@@ -0,0 +1,336 @@
+			},
+			WorkerProfiles: &[]redhatopenshift.WorkerProfile{
+				{
+					Name:       to.StringPtr("worker"),
+					Count:      to.Int32Ptr(3),
+					DiskSizeGB: to.Int32Ptr(128),
+					VMSize:     redhatopenshift.VMSize1StandardD2sV3,
+					SubnetID:   to.StringPtr(fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s", i.conf.AzureSubscriptionID, i.conf.AROResourceGroup, vnetName, wsName)),
+				},
+			},
+			ApiserverProfile: &redhatopenshift.APIServerProfile{
+				Visibility: redhatopenshift.Public,
+			},
+			IngressProfiles: &[]redhatopenshift.IngressProfile{
+				{
+					Visibility: redhatopenshift.Visibility1Public,
+					Name:       to.StringPtr("default"),
+				},
+			},
+		},
+	})
+
+	return err
+}
+
+func (i *Infrastructure) deleteCluster(ctx context.Context) error {
+	return i.aroClientSet.DeleteAndWait(ctx, i.conf.AROResourceGroup, i.conf.Cluster+"-"+i.conf.AROResourceGroup+"-"+i.conf.RandSuffix)
+}
+
+func (i *Infrastructure) DeployResources() error {
+
+	rgName := i.conf.AROResourceGroup
+
+	rtName := "ci-routetable"
+	vnetName := "dev-vnet"
+	msName := "master-subnet"
+	wsName := "worker-subnet"
+
+	ctx := context.Background()
+
+	result, err := i.appClientSet.GetServicePrincipalsIDByAppID(ctx, FP_CLIENT_ID)
+	if err != nil {
+		return err
+	}
+
+	fpSP := *result.Value
+
+	err = i.deployResourceGroup(ctx)
+	if err != nil {
+		return err
+	}
+
+	appID, appOID, appPass, spID, err := i.deployActiveDirectory(ctx)
+	if err != nil {
+		return err
+	}
+
+	i.appOID = appOID
+
+	t := arm.Template{
+		Schema:         "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+		ContentVersion: "1.0.0.0",
+		Resources: []*arm.Resource{
+			networkRouteTable(rtName),
+			networkVnet(vnetName, rtName, msName, wsName),
+			rbacResourceGroupRoleAssignment(spID, rgName),
+			rbacVnetRoleAssignment(spID, vnetName, "sp-client"),
+			rbacVnetRoleAssignment(fpSP, vnetName, "fp-client"),
+			rbacRouteTableRoleAssignment(spID, rtName, "sp-client"),
+			rbacRouteTableRoleAssignment(fpSP, rtName, "fp-client"),
+		},
+	}
+
+	err = i.deployARMTemplate(ctx, &t)
+	if err != nil {
+		return err
+	}
+
+	err = i.deployCluster(ctx, appID, appPass, vnetName, msName, wsName)
+
+	return nil
+}
+
+func (i *Infrastructure) DeleteResources() error {
+	ctx := context.Background()
+
+	rgName := i.conf.AROResourceGroup
+	aroName := i.conf.Cluster
+
+	err := i.deleteCluster(ctx)
+	if err != nil {
+		return err
+	}
+
+	// wait for 300s until the cluster deletion is finished
+	// TODO: is this sufficient?
+	for r := 0; r < 10; r++ {
+		_, err := i.aroClientSet.Get(ctx, rgName, aroName)
+		if detailedErr, ok := err.(autorest.DetailedError); ok &&
+			detailedErr.StatusCode == http.StatusNotFound {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		time.Sleep(30 * time.Second)
+	}
+
```

Comment:
deleteCluster already waits for deletion, why this is needed?

Reply:
I thought the same. The behavior is different. It waits for the response, which is returned immediately once the delete process begins. The deletion is initiated as an async operation.